kv/kvserver: TestLeaseQueueShedsOnIOOverload failed

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/4fd5bdf6-3b60-4551-ad0a-0177ca687dc0?testReportRun=1&testReportShard=5&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [31e84cb3a57c52a779ff0982c95fb26646b54926](https://github.com/cockroachdb/cockroach/commits/31e84cb3a57c52a779ff0982c95fb26646b54926):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload469026051
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 1
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload469026051
--- FAIL: TestLeaseQueueShedsOnIOOverload (238.61s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=5</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-46414

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/3342d016-bbad-434c-9414-3726f16e028b?testReportRun=2&testReportShard=5&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [758fe6af0492e04f677196745f28a546b3a657cc](https://github.com/cockroachdb/cockroach/commits/758fe6af0492e04f677196745f28a546b3a657cc):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload4050805002
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 1
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload4050805002
--- FAIL: TestLeaseQueueShedsOnIOOverload (237.22s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=5</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/3da25971-51ee-4d08-b8ca-5ff9827f8ce8?testReportRun=3&testReportShard=5&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [758fe6af0492e04f677196745f28a546b3a657cc](https://github.com/cockroachdb/cockroach/commits/758fe6af0492e04f677196745f28a546b3a657cc):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload1024607057
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 26
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload1024607057
--- FAIL: TestLeaseQueueShedsOnIOOverload (271.52s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=5</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/c7d27b63-8e97-4b55-aae6-260fa490ee89?testReportRun=2&testReportShard=5&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [6647d059a3872cee318ba35bda885938790b7c80](https://github.com/cockroachdb/cockroach/commits/6647d059a3872cee318ba35bda885938790b7c80):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload2702271689
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 1
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload2702271689
--- FAIL: TestLeaseQueueShedsOnIOOverload (238.57s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=5</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/98c86f20-fa6d-4ff3-a75d-d7314f84783c?testReportRun=1&testReportShard=37&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [a616c80e5c69c33ab1df58473eb0c3d0c522df36](https://github.com/cockroachdb/cockroach/commits/a616c80e5c69c33ab1df58473eb0c3d0c522df36):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload1046702761
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 1
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload1046702761
--- FAIL: TestLeaseQueueShedsOnIOOverload (236.89s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=1</code>
 - <code>shard=37</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/023a3a14-949f-4177-868c-725e04f16957?testReportRun=22&testReportShard=37&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [c25e426bb34ed6230a863fa94555dd0e1f60524b](https://github.com/cockroachdb/cockroach/commits/c25e426bb34ed6230a863fa94555dd0e1f60524b):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload2469071243
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 1
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload2469071243
--- FAIL: TestLeaseQueueShedsOnIOOverload (55.74s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=22</code>
 - <code>shard=37</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/0bdefeed-2600-4def-a0ae-a98c62b60969?testReportRun=3&testReportShard=37&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [26def9e9056d9ef4c866e2237f36d0784f409a46](https://github.com/cockroachdb/cockroach/commits/26def9e9056d9ef4c866e2237f36d0784f409a46):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload4078186991
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 1
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload4078186991
--- FAIL: TestLeaseQueueShedsOnIOOverload (239.33s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=37</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/b4ccd550-831b-41a6-bf25-729cb168ef7b?testReportRun=3&testReportShard=37&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [32ef1302b3c6b67909a7411245ee37b3049c7b2a](https://github.com/cockroachdb/cockroach/commits/32ef1302b3c6b67909a7411245ee37b3049c7b2a):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload1834117486
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 1
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload1834117486
--- FAIL: TestLeaseQueueShedsOnIOOverload (236.76s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=37</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/907170eb-9ae8-49c2-9398-dd0f45b7c31f?testReportRun=3&testReportShard=37&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [ecfb74b7ad75327f11814966a2cdab1b9793a549](https://github.com/cockroachdb/cockroach/commits/ecfb74b7ad75327f11814966a2cdab1b9793a549):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload4076862335
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 1
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload4076862335
--- FAIL: TestLeaseQueueShedsOnIOOverload (236.78s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=37</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/b0317590-ee19-4932-ab72-6a3e23ee3f8a?testReportRun=2&testReportShard=37&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [223ea1e063897fface6fcb639760554ed5289b3a](https://github.com/cockroachdb/cockroach/commits/223ea1e063897fface6fcb639760554ed5289b3a):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload626322313
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 1
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload626322313
--- FAIL: TestLeaseQueueShedsOnIOOverload (239.02s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=37</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

Repros in about 5 minutes with:

```bash
dev test pkg/kv/kvserver -f TestLeaseQueueShedsOnIOOverload --stress --deadlock
```


<details><summary>Details</summary>
<p>

```bash
==================== Test output for //pkg/kv/kvserver:kvserver_test (run 89 of 1000):
<jemalloc>: No THP support: thp:never
<jemalloc>: option background_thread currently supports pthread only
<jemalloc>: No THP support: thp:never
<jemalloc>: option background_thread currently supports pthread only
metamorphic: use COCKROACH_RANDOM_SEED=2632494621290179942 for reproduction
initialized metamorphic constant "max-batch-size" with value 7511
initialized metamorphic constant "max-batch-byte-size" with value 28393104
initialized metamorphic constant "backup-merge-file-buffer-size" with value 14514227
initialized metamorphic constant "parse-json-impl" with value 1
initialized metamorphic constant "coldata-batch-size" with value 3175
initialized metamorphic constant "COCKROACH_ENABLE_HDR_HISTOGRAMS" with value true
initialized metamorphic constant "default-batch-bytes-limit" with value 9560
initialized metamorphic constant "kvadmission.flow_control.mode" with value apply_to_elastic
initialized metamorphic constant "spilling-queue-initial-len" with value 2
initialized metamorphic constant "merge-joiner-groups-buffer" with value 4
initialized metamorphic constant "rpc.batch_stream_pool.enabled" with value false
initialized metamorphic constant "distsender-leaseholder-first" with value false
initialized metamorphic constant "kv.transaction.randomized_anchor_key.enabled" with value true
initialized metamorphic constant "lock-table-iters-before-seek" with value 1
initialized metamorphic constant "mvcc-incremental-iter-tbi" with value false
initialized metamorphic constant "storage.experimental.eventually_file_only_snapshots.enabled" with value false
initialized metamorphic constant "kv.snapshot_receiver.excise.enabled" with value false
initialized metamorphic constant "storage.columnar_blocks.enabled" with value false
initialized metamorphic constant "mvcc-max-iters-before-seek" with value 2
initialized metamorphic constant "disable-checksstconflicts-range-key-masking" with value true
initialized metamorphic constant "storage.ingestion.value_blocks.enabled" with value false
initialized metamorphic constant "addsst-rewrite-concurrency" with value 5
initialized metamorphic constant "addsst-rewrite-forced" with value true
initialized metamorphic constant "changefeed.event_consumer_worker_queue_size" with value 3
initialized metamorphic constant "ColIndexJoin-batch-size" with value 2846474
initialized metamorphic constant "ColIndexJoin-using-streamer-batch-size" with value 4650279
initialized metamorphic constant "parallel-scan-result-threshold" with value 7369
initialized metamorphic constant "inverted-joiner-batch-size" with value 1
initialized metamorphic constant "kv.split.estimated_mvcc_stats.enabled" with value false
initialized metamorphic constant "kv.snapshot.ingest_as_write_threshold" with value 1073741824
initialized metamorphic constant "kv.raft.leader_fortification.fraction_enabled" with value 1
initialized metamorphic constant "kv.snapshot_rebalance.max_sst_size" with value 235840331
initialized metamorphic constant "conn-executor-force-exec-with-pausable-portal" with value true
initialized metamorphic constant "copy-batch-size" with value 30587
initialized metamorphic constant "async-IE-result-channel-buffer-size" with value 5
initialized metamorphic constant "use-index-lookup-for-descriptors-in-database" with value false
initialized metamorphic constant "tail-call-optimization-enabled" with value false
initialized metamorphic constant "multiple_active_portals_enabled" with value true
initialized metamorphic constant "backup.write_metadata_with_external_ssts.enabled" with value false
initialized metamorphic constant "export_request_verbose_tracing" with value true
initialized metamorphic constant "restore-worker-concurrency" with value 1
initialized metamorphic constant "logical_replication.consumer.use_implicit_txns.enabled" with value false
initialized metamorphic constant "logical_replication.consumer.try_optimistic_insert.enabled" with value false
initialized metamorphic constant "default_range_distribution_strategy" with value default
initialized metamorphic constant "changefeed.parallel_io.request_quota" with value 16
I250210 20:41:46.818816 1 (gostd) rand.go:256  [-] 1  random seed: 2632494621290179942
test logs left over in: /tmp/cockroach/_tmp/76d155434b67ec2ece133e41f93c7b73/logTestLeaseQueueShedsOnIOOverload3653864004
--- FAIL: TestLeaseQueueShedsOnIOOverload (239.57s)
    test_log_scope.go:165: test logs captured to: /tmp/cockroach/_tmp/76d155434b67ec2ece133e41f93c7b73/logTestLeaseQueueShedsOnIOOverload3653864004
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 1
    panic.go:626: -- test log scope end --
FAIL
I250210 20:45:46.413524 1 (gostd) testmain.go:1715  [-] 1  Test //pkg/kv/kvserver:kvserver_test exited with error code 1
```

</p>
</details>

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/b8448e7a-83ad-42eb-9503-3e98333881ae?testReportRun=2&testReportShard=39&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [8c5af013aa725d4f441abeff282894ff22f04ba7](https://github.com/cockroachdb/cockroach/commits/8c5af013aa725d4f441abeff282894ff22f04ba7):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload410400118
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 1
    panic.go:629: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload410400118
--- FAIL: TestLeaseQueueShedsOnIOOverload (239.04s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=39</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/b3253065-dc87-4641-9905-e3981bd8c3a8?testReportRun=1&testReportShard=40&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [f54cc8e9f0b58664b1fa6db15272889eee8ead41](https://github.com/cockroachdb/cockroach/commits/f54cc8e9f0b58664b1fa6db15272889eee8ead41):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload3063032539
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 21
    panic.go:629: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload3063032539
--- FAIL: TestLeaseQueueShedsOnIOOverload (256.00s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=40</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/e6e361e5-57e0-48da-b9d8-3ab11474ea0a?testReportRun=3&testReportShard=40&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [540a0423c37e471e51012b325ec822c03ccb1067](https://github.com/cockroachdb/cockroach/commits/540a0423c37e471e51012b325ec822c03ccb1067):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload286895561
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 27
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload286895561
--- FAIL: TestLeaseQueueShedsOnIOOverload (240.99s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=3</code>
 - <code>shard=40</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/c9e2ef8d-70c5-4adf-a124-6ac037a1ffa1?testReportRun=1&testReportShard=41&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [b53856cc8cebf5fe28a8fca3940f5603b5b2d833](https://github.com/cockroachdb/cockroach/commits/b53856cc8cebf5fe28a8fca3940f5603b5b2d833):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload1949897996
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 27
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload1949897996
--- FAIL: TestLeaseQueueShedsOnIOOverload (256.96s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=41</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/92787237-f97d-4a75-ae51-0d7adfb6b742?testReportRun=1&testReportShard=44&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [11289170e6103a1b5e0c1083ecd11f4e889264f6](https://github.com/cockroachdb/cockroach/commits/11289170e6103a1b5e0c1083ecd11f4e889264f6):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload814983504
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 26
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload814983504
--- FAIL: TestLeaseQueueShedsOnIOOverload (269.73s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=44</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestLeaseQueueShedsOnIOOverload [failed](https://tanzanite.cluster.engflow.com/invocations/default/1608ce5d-bdf0-4b64-a33b-bdc5db4a772f?testReportRun=1&testReportShard=47&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [342ea3423ddd4e9a84c598264fef733c0e2ae076](https://github.com/cockroachdb/cockroach/commits/342ea3423ddd4e9a84c598264fef733c0e2ae076):


```
=== RUN   TestLeaseQueueShedsOnIOOverload
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestLeaseQueueShedsOnIOOverload1833880168
    test_log_scope.go:76: use -show-logs to present logs inline
    lease_queue_test.go:560: condition failed to evaluate within 3m45s: from lease_queue_test.go:566: expected 0 leases on store 1, found 28
    panic.go:635: -- test log scope end --
test logs left over in: outputs.zip/logTestLeaseQueueShedsOnIOOverload1833880168
--- FAIL: TestLeaseQueueShedsOnIOOverload (289.62s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>race=true</code>
 - <code>run=1</code>
 - <code>shard=47</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #139324 kv/kvserver: TestLeaseQueueShedsOnIOOverload failed [C-test-failure O-robot P-2 T-kv X-duplicate branch-release-25.1 no-test-failure-activity]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLeaseQueueShedsOnIOOverload.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvserver/lease_queue_test.go
+++ b/pkg/kv/kvserver/lease_queue_test.go
@@ -544,6 +544,11 @@ func TestLeaseQueueRaceReplicateQueue(t *testing.T) {
 func TestLeaseQueueShedsOnIOOverload(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
+
+	// The SucceedsSoon has been observed to occasionally time out in both
+	// deadlock and race builds.
+	skip.UnderDuressWithIssue(t, 138903)
+
 	ctx := context.Background()
 
 	tc := testcluster.StartTestCluster(t, 3, base.TestClusterArgs{})

